### PR TITLE
program: rewrite crate tests in mollusk

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR rewrites the tests in the program's `tests` folder that use `solana-program-test` to use [Mollusk](https://github.com/buffalojoec/mollusk).

There are more tests in the processor that can be handled in a subsequent PR.

### CODE REVIEW

File: program/tests/close_account.rs [hunk 1]
```diff
@@ -1,200 +1,93 @@
...
         &[
-            instruction::close_account(
-                &token_program_id,
-                &token_account.pubkey(),
-                &destination,
-                &owner.pubkey(),
-                &[],
-            )
-            .unwrap(),
+            instruction::close_account(&spl_token::id(), &account, &owner, &owner, &[]).unwrap(),
```
Comment (joncinque) [inline program/tests/close_account.rs:32]: The previous version of this test had a few more checks -- can this retain the `destination` account that receives the lamports from close and check that `destination` has lamports?

**Review by joncinque — COMMENTED** [review]
Looks really great! The red lines on the Cargo.lock are a beautiful sight to behold. Just the one question about the close test

File: program/tests/close_account.rs (hunk 1 above)
Comment (buffalojoec) [reply program/tests/close_account.rs]: Done!

File: program/Cargo.toml [hunk 2]
```diff
@@ -22,9 +22,9 @@ thiserror = "2.0"
 
 [dev-dependencies]
 lazy_static = "1.5.0"
+mollusk-svm = "0.0.13"
 proptest = "1.5"
 serial_test = "3.2.0"
-solana-program-test = "2.1.0"
```
Comment (buffalojoec) [inline program/Cargo.toml:27]: I can't believe dropping this single dependency sheds 4,143 lines from the lockfile. 😅

**Review by joncinque — APPROVED** [review]
Beautiful, thanks!

**Pull request merged**